<!DOCTYPE html>
<html lang="en-US">

<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <title>Exploring HA Deep Mutational Scan Data</title>

  <script src="https://d3js.org/d3.v5.min.js"></script>
  <script src="https://d3js.org/d3-array.v2.min.js"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/d3-legend/2.25.6/d3-legend.js"></script>
  <script src="https://www.ncbi.nlm.nih.gov/Structure/icn3d/lib/jquery.min.js"></script>
  <script src="https://www.ncbi.nlm.nih.gov/Structure/icn3d/lib/jquery-ui.min.js"></script>
  <script src="https://www.ncbi.nlm.nih.gov/Structure/icn3d/lib/three.min.js"></script>
  <script src="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d_full_ui.min.js"></script>
  <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css" integrity="sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T" crossorigin="anonymous">
  <link rel="stylesheet" href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/lib/jquery-ui.min.css" />
  <link rel="stylesheet" href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d_full_ui.css" />
  <link rel="stylesheet" href="styles.css" />
</head>

<body>
  <div class="container">
    <div class="row mt-3">
      <div class="col-12">
        <h1>Seasonal influenza evolution in the lab and nature</h1>

        <p>The line plot below reveals sites in seasonal influenza A/H3N2's hemagglutinin where mutations have allowed individual viruses to escape human adaptive immunity in laboratory conditions.</p>

        <p>Click on a site shown on the line plot to see the position of that site in the three-dimensional protein structure of hemagglutinin and the frequencies of mutations at that site in natural influenza populations.</p>

        <p>Sites shown in <strong style="color: #000000">black</strong> have frequency data. Sites shown in <strong style="color: #999999">grey</strong> do not have any genetic variation in natural populations and therefore do not have any mutation
          frequency data.</p>
      </div>
    </div>

    <div class="row mb-5 border-top border-bottom pt-3 pb-3">
      <div class="col-8">
        <div id="line_plot" class="row"></div>
        <div id="frequencies" class="row"></div>
      </div>
      <div class="col-4">
        <div class="list-group list-group-horizontal-sm mb-1">
          <button type="button" id="save_image" class="list-group-item list-group-item-action flex-fill"><strong style="color: #000000">Save Image</strong></button>
          <button type="button" id ="remove_selections" class="list-group-item list-group-item-action flex-fill"><strong style="color: #000000">Clear Selections</strong></button>
          <button type="button" id="send_state" class="list-group-item list-group-item-action flex-fill"><strong style="color: #000000">Share Link</strong></button>
          <button type="button" id="load_command" class="list-group-item list-group-item-action flex-fill" data-toggle="modal" data-target="#myModal"><strong style="color: #000000">Toggle Commandline</strong></button>
      </div>

        <div id="icn3dwrap" class="row"></div>
        <div id="colorKeydiv" class="row"></div>
      </div>


    </div>

    <div class="row mt-3 mb-3">
      <div class="col-12">
        <h2>Background on person-to-person variation in influenza escape mutations</h2>

        <p>The data shown in the top left panel of this visualization are based on recent lab experiments from <a href="https://research.fhcrc.org/bloom/en.html">Jesse Bloom's lab</a> at the Fred Hutchinson Cancer Research Center. These experiments were led by Juhye Lee, a doctoral student in the Medical Science Training Program (MSTP) and the Genome Science department at the University of Washington.</p>

        <p>Seasonal influenza A/H3N2 viruses evolve rapidly, accumulating amino acid mutations in their surface proteins that allow them to escape detection by antibodies from our previous infections or vaccinations. The surface protein hemagglutinin is a primary target of our immune systems. To design more effective vaccines for influenza, we need to know which positions in hemagglutinin can mutate to enable escape from antibodies and whether circulating influenza viruses have any genetic variation at those positions.</p>

        <p>To this end, Juhye created every possible single amino acid mutation to the hemagglutinin gene of a specific influenza virus. She infected cells with these mutant viruses in the presence of human sera with influenza antibodies. Juhye measured the number of times each mutation allowed a virus to escape human antibodies relative to the number of times those mutations allowed viruses to infect cells in the absence of antibodies. The line plot below shows this relative enrichment for antibody escape from one human's serum for each position in hemagglutinin. The three-dimensional model to the right shows the positions of these enrichment values protein structure.</p>

        <div class="row">
          <div class="col-4">
            <img src="antigenic-escape-line-plot.png" style="width: 300px" />
          </div>
          <div class="col-3">
            <img src="antigenic-escape-on-protein-structure.png" style="width: 150px" />
          </div>
        </div>

        <p>Juhye identified important positions of hemagglutinin from the figures above and inspected the frequency of mutations at these sites in natural influenza populations. The stream plot below shows an example of mutation frequencies at one position in hemagglutinin over the last 12 years. The presence of natural genetic variation at positions identified in laboratory experiments supports their importance for the evolutionary success of influenza viruses.</p>

        <p><img src="antigenic-escape-mutants-in-nature.png" style="width: 300px" /></p>

        <p>The interactive visualization on this page allows Juhye and others to explore these escape mutations interactively by selecting individual positions in hemagglutinin and displaying the frequencies of mutations at the selected site and the position of the site in the three-dimensional structure of the hemagglutinin protein.</p>
      </div>
    </div>

    <!-- warning modal before reloading with a command line -->
    <div class="modal" id="myModal" tabindex="1" role="dialog" aria-hidden="true">
    <div class="modal-dialog modal-sm" role="document">
    <div class="modal-content">
    <div class="modal-header">
    <h4 id="myModalLabel"><font color="white">Clear all selections and open the interactive command line.</font></h4>

    </div>
    <div class="modal-body">
    <p><strong style="color: red">WARNING!</strong> Clicking confirm will remove all current selections and load a new view. Are you sure you want to continue?</p>
    </div>
    <div class="modal-footer">
    <button type="confirm" id = "cancel" class="btn btn-warning" data-dismiss="modal">Close</button>
    <button type="cancel" id = "confirm" class="btn btn-danger">Yes Reload</button>
    </div>
    </div>
    </div>
    </div>
  </div>
  <script src="_javascript/color-legend.js"></script>
  <script src="_javascript/line_plot_zoom.js" type="text/javascript"></script>
  <script src="_javascript/frequencies.js" type="text/javascript"></script>
  <script src="_javascript/prot_struct.js"></script>
  <script src="_javascript/modal_activation.js" type="text/javascript"></script>
</body>


</html>
